### PR TITLE
Add support for multiple state classes

### LINKED ISSUE #157: Supporting multiple initial/top/pinned/unpinned class tokens?
```js
{
  // ...
  classes: {
    initial: 'animated hinge',
    // ...
  }
}
```

throws:

```
Uncaught DOMException: Failed to execute 'add' on 'DOMTokenList':
The token provided ('a b') contains HTML space characters, which are not valid in tokens.
```

cause the source code doesn't support multiple class tokens:

```js
this.elem.classList.add(this.classes.initial);
this.elem.classList.remove(classes.unpinned, classes.pinned, classes.top, classes.initial);
```

How about we improving it?

### PR DESCRIPTION
**Pull request by hacknug**
As discussed in that issue, this PR allows users to specify multiple classes on their Headroom options using a space-delimited string.

I used the spread operator but I'm not sure if we're using Babel or not. Can rewrite it using `.apply()` if not but maybe it would be better to introduce new tolling so we can start writing modern JS for v1.0 👍 

Closes #157.

### CODE REVIEW

**Comment by WickyNilliams** [comment]
Yes can you switch to apply for now, there's no babel or anything at the moment, everything is ES5. Also the build has failed due to linting issues

**Comment by hacknug** [comment]
All good now. Had to add `eslint` to `devDependencies` to make it work on my side (guessing you have it installed globally). Also added `eslint-plugin-cypress` to prevent linting issues on the tests 👍

File: src/Headroom.js [hunk 1]
```diff
@@ -160,11 +160,17 @@ Headroom.prototype = {
...
 
   removeClass: function(className) {
-    this.elem.classList.remove(this.classes[className]);
+    this.elem.classList.remove.apply(
+      this.elem.classList,
+      this.classes[className].split(" ")
+    );
   },
 
   hasClass: function(className) {
```
Comment (WickyNilliams) [inline src/Headroom.js]: We also need changes to `hasClass`, so it checks that each of the (multiple) classes is present. If you add a test like this you will see the current implementation fails:

```js
  it("handles multiple classes", () => {
    initialiseHeadroom({
      classes: {
        pinned: "headroom--pinned foo",
        unpinned: "headroom--unpinned bar"
      }
    });

    cy.scrollTo(0, 50);
    cy.get("header")
      .should("not.be.pinned")
      .should("have.class", "bar");

    cy.scrollTo(0, 25);
    cy.get("header")
      .should("be.pinned")
      .should("have.class", "foo");
  });
```

But if you change to this, you should see the tests now pass:

```js
  hasClass: function(className) {
    return this.classes[className].split(" ").every(function(cls) {
      return this.classList.contains(cls);
    }, this.elem);
  },
```

**Comment by WickyNilliams** [comment]
Good spot on the eslint installation. I think it's implicitly installed by `rollup-plugin-eslint`, it should really be a peerDependency of that library, but for some reason it is. Better to be explicit anyway :)

File: src/Headroom.js (hunk 1 above)
Comment (hacknug) [reply src/Headroom.js]: `hasClass` is only used by the library itself and always with a single class. Do we really need to touch it?

Comment (WickyNilliams) [reply src/Headroom.js]: Yes, it's a required change. Can you add the test above to this PR? You will see the failure then, and we can catch any future regressions.

The reason being that (unfortunately) some state is stored in the DOM, and we use `hasClass` to check whether a state transition has occurred (e.g. pinned -> unpinned). Without modifying `hasClass`, if you supply multiple classes then a call like `this.elem.classList.contains("unpinned foo bar")` would be made, which would return false. Thus we need to check for the existence of every class for each state.

**Comment by WickyNilliams** [comment]
Any updates on this @hacknug?

**Comment by WickyNilliams** [comment]
I think this branch could actually be merged to `master` since it is a non-breaking change? Would save you rebasing etc then. I'm happy to do that on the `v1` branch

File: src/Headroom.js (hunk 1 above)
Comment (hacknug) [reply src/Headroom.js]: Just pushed a couple commits with the changes

**Comment by hacknug** [comment]
@WickyNilliams sorry but I took some days off open-source due to the protests and everything going on here in Barcelona. Had to catch up with client work after that and a few days of vacations.

Just pushed the changes you requested and changed the target branch 👍

**Comment by WickyNilliams** [comment]
No worries, thanks for coming back to finish it off :) I think the only issue now is that the branch needs a `git rebase master`. There are some changes in the diff which are from the v1 branch, so I think rebasing will fix that

File: src/Headroom.js [hunk 2]
```diff
@@ -35,18 +35,10 @@ Headroom.prototype = {
...
-            { offset: self.offset, tolerance: self.tolerance },
-            self.update.bind(self)
-          );
-        },
-        100,
-        this
+      this.scrollTracker = trackScroll(
+        this.scroller,
+        { offset: this.offset, tolerance: this.tolerance },
+        this.update.bind(this)
```
Comment (WickyNilliams) [inline src/Headroom.js]: E.g. this change is from `v1`, and isn't yet on `master`

**Comment by WickyNilliams** [comment]
I've created a new PR to supersede this one. I have cherry picked your commits so that you still get the authorship in the history :)

**Pull request closed without merging**